### PR TITLE
Update format to have J and W and implement json always on xml models

### PR DESCRIPTION
**Pull request by m-nash**
Addresses this comment https://apiview.dev/Assemblies/Review/52b58c3eb11144e383612c92c69f1f74/6f9d249df11249928f1caf8b7226151f?diffRevisionId=c53fffb104884669addf0e6c1fdfd4d2&doc=False&diffOnly=True#b771bc223886485ea538173f1b1b8794

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/core/Azure.Core/src/Serialization/ModelSerializerFormat.cs [hunk 1]
```diff
@@ -11,13 +11,13 @@ namespace Azure.Core.Serialization
...
     {
-        internal const string DataValue = "D";
+        internal const string JsonValue = "J";
         internal const string WireValue = "W";
 
         /// <summary>
         /// Specifies the data format where IgnoreReadOnly and IgnoreAdditionalProperties are false.
         /// </summary>
-        public static readonly ModelSerializerFormat Data = new ModelSerializerFormat(DataValue);
+        public static readonly ModelSerializerFormat Json = new ModelSerializerFormat(JsonValue);
```
Comment (christothes) [inline sdk/core/Azure.Core/src/Serialization/ModelSerializerFormat.cs:20]: Will the doc string here make sense to most users of this API? Reading it without additional context, I'm not exactly sure what it means.

File: sdk/core/Azure.Core/src/Serialization/ModelSerializerOptions.cs [hunk 2]
```diff
@@ -12,32 +12,28 @@ namespace Azure.Core.Serialization
...
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ModelSerializerOptions" /> class.
+        /// </summary>
+        /// <param name="format"> The format to serialize to and deserialize from. </param>
+        public ModelSerializerOptions(ModelSerializerFormat format)
         {
-            //throw ArgumentException if not "D" or "W"
-            Format = ValidateFormat(format);
+            Format = format;
```
Comment (christothes) [inline sdk/core/Azure.Core/src/Serialization/ModelSerializerOptions.cs:26]: It still would be possible to pass a random value for format here - should we leave the validation in place?

Comment (m-nash) [reply sdk/core/Azure.Core/src/Serialization/ModelSerializerOptions.cs:26]: random value is by design so people who implement our interface can come up with their own formats.  Each model will throw if it receives a format it doesn't understand.

File: sdk/core/Azure.Core/src/Serialization/ModelSerializerFormat.cs (hunk 1 above)
Comment (m-nash) [reply sdk/core/Azure.Core/src/Serialization/ModelSerializerFormat.cs:20]: We are address doc comments in a different PR, but yes these are not final

**Review by christothes — APPROVED** [review]

**Pull request merged**